packages/purpose-process/src/services/purposeService.ts

```diff
@@ -399,79 +392,62 @@ export function purposeServiceBuilder(
       });
       await repository.createEvent(event);
     },
-    async updatePurpose({
-      purposeId,
-      purposeUpdateContent,
-      organizationId,
-      correlationId,
-      logger,
-    }: {
-      purposeId: PurposeId;
-      purposeUpdateContent: purposeApi.PurposeUpdateContent;
-      organizationId: TenantId;
-      correlationId: CorrelationId;
-      logger: Logger;
-    }): Promise<{ purpose: Purpose; isRiskAnalysisValid: boolean }> {
+    async updatePurpose(
+      purposeId: PurposeId,
+      purposeUpdateContent: purposeApi.PurposeUpdateContent,
+      { authData, correlationId, logger }: WithLogger<AppContext>
+    ): Promise<{ purpose: Purpose; isRiskAnalysisValid: boolean }> {
       logger.info(`Updating Purpose ${purposeId}`);
       return await performUpdatePurpose(
         purposeId,
         {
           updateContent: purposeUpdateContent,
           mode: eserviceMode.deliver,
         },
-        organizationId,
+        authData,
         readModelService,
         correlationId,
         repository
       );
     },
-    async updateReversePurpose({
-      purposeId,
-      reversePurposeUpdateContent,
-      organizationId,
-      correlationId,
-      logger,
-    }: {
-      purposeId: PurposeId;
-      reversePurposeUpdateContent: purposeApi.ReversePurposeUpdateContent;
-      organizationId: TenantId;
-      correlationId: CorrelationId;
-      logger: Logger;
-    }): Promise<{ purpose: Purpose; isRiskAnalysisValid: boolean }> {
+    async updateReversePurpose(
+      purposeId: PurposeId,
+      reversePurposeUpdateContent: purposeApi.ReversePurposeUpdateContent,
+      { authData, correlationId, logger }: WithLogger<AppContext>
+    ): Promise<{ purpose: Purpose; isRiskAnalysisValid: boolean }> {
       logger.info(`Updating Reverse Purpose ${purposeId}`);
       return await performUpdatePurpose(
         purposeId,
         {
           updateContent: reversePurposeUpdateContent,
           mode: eserviceMode.receive,
         },
-        organizationId,
+        authData,
         readModelService,
         correlationId,
         repository
       );
     },
-    async deletePurpose({
-      purposeId,
-      organizationId,
-      correlationId,
-      logger,
-    }: {
-      purposeId: PurposeId;
-      organizationId: TenantId;
-      correlationId: CorrelationId;
-      logger: Logger;
-    }): Promise<void> {
+    async deletePurpose(
+      purposeId: PurposeId,
+      { authData, correlationId, logger }: WithLogger<AppContext>
+    ): Promise<void> {
       logger.info(`Deleting Purpose ${purposeId}`);
 
       const purpose = await retrievePurpose(purposeId, readModelService);
 
-      assertOrganizationIsAConsumer(organizationId, purpose.data.consumerId);
-
       if (!isDeletable(purpose.data)) {
         throw purposeCannotBeDeleted(purpose.data.id);
       }
 
+      assertRequesterCanActAsConsumer(
```

Comment:
Moved the check after the isDeletable in order to prevent the retrieve from the readModel when not needed